src/Suave.Tests/Owin.fs

```diff
@@ -148,5 +159,177 @@ let endToEnd =
+          |> Encoding.UTF8.GetString
+        match credentials.Split([|':'|]) with
+        | [|"foo";"bar"|] ->
+          let task = next.Invoke(env)
+          task
+        | _ ->
+          env.[OwinConstants.responseStatusCode] <- box 401
+          Threading.Tasks.Task.FromResult() :> Threading.Tasks.Task
+        ))
+
+      let composedApp =
+        OwinApp.ofAppFunc (basicAuthMidFunc.Invoke(noContent))
+
+      let asserts (result : HttpResponseMessage) =
+        eq "Http Status Code" HttpStatusCode.NoContent result.StatusCode
+        eq "Reason Phrase set by server" "No Content" result.ReasonPhrase
+
+      let sendAuthHeader (req : HttpRequestMessage) =
+        req.Headers.Authorization <- Net.Http.Headers.AuthenticationHeaderValue("Basic", Convert.ToBase64String("foo:bar"B))
+        req
+
+      runWithConfig composedApp |> reqResp Types.GET "/" "" None None DecompressionMethods.GZip sendAuthHeader asserts
+
+    testCase "Setting a path in Suave should mount an OWIN app at that path and set requestPathBase" <| fun _ ->
+      let ok (env : OwinEnvironment) =
+        let requestPathBase : string = unbox env.[OwinConstants.requestPathBase]
+        printfn "owin.RequestPathBase is %s" requestPathBase
+        //eq "owin.RequestPathBase" requestPathBase "/owin"
+        let requestPath : string = unbox env.[OwinConstants.requestPath]
+        printfn "owin.RequestPath is %s" requestPath
+        if requestPath <> "/app" then
+          env.[OwinConstants.responseStatusCode] <- box 404
+        else () // 200 OK
+        async.Return ()
+
+      let composedApp =
+        pathRegex "/owin(/.+)*" >>= OwinApp.ofApp ok
+
+      let asserts (result : HttpResponseMessage) =
+        eq "Http Status Code" HttpStatusCode.OK result.StatusCode
+        eq "Reason Phrase set by server" "OK" result.ReasonPhrase
+
+      runWithConfig composedApp |> reqResp Types.GET "/owin/app" "" None None DecompressionMethods.GZip id asserts
+
+    testCase "Manually mount an OWIN app at that path and set requestPathBase" <| fun _ ->
+      let ok (env : OwinEnvironment) =
+        let requestPathBase : string = unbox env.[OwinConstants.requestPathBase]
+        printfn "owin.RequestPathBase is %s" requestPathBase
+        //eq "owin.RequestPathBase" requestPathBase "/owin"
+        let requestPath : string = unbox env.[OwinConstants.requestPath]
+        printfn "owin.RequestPath is %s" requestPath
+        if requestPath <> "/app" then
+          env.[OwinConstants.responseStatusCode] <- box 404
+        else () // 200 OK
+        async.Return ()
+
+      let composedApp =
+        pathScan "/owin/%s" (fun path -> OwinApp.ofApp (fun env -> async {
+          env.[OwinConstants.requestPathBase] <- box "/owin"
+          env.[OwinConstants.requestPath] <- box ("/" + path)
+          do! ok env
+          env.[OwinConstants.requestPathBase] <- box ""
+          env.[OwinConstants.requestPath] <- box ("/owin/" + path)
+          })
```

Comment:
The above shows what would need to occur in order to map the `path` to the `owin.RequestPathBase`. I think `path` in Suave is meant to be used differently, so perhaps this could be moved to the Owin module?